cli/cli_test.go

```diff
@@ -35,6 +35,16 @@ type cliTest struct {
 	*server.TestServer
 }
 
+// Stop the server, or wait until stopping is done
+// if it was already initiated by the caller.
+func (c *cliTest) cleanup(wasStopped *bool) {
+	if *wasStopped {
```

Comment:
There is no `t.FailNow()` in `Example` tests, although that's a concern if there are any non-example CLI tests. You could pass the `t` object (or nil) into cleanup so that you could check `t.Failed()`.